### ISSUE TITLE
building PDF documentation would cause FTBFS

### ISSUE THREAD

**Issue by hosiet** [d1vanov/QEverCloud#8]
Doxygen is set to build latex documentations by default. However, this would only output latex files + Makefile.

Building `refman.pdf` with make would FTBFS. Unfortunately the error log is lost, but the problem is due to pdflatex+ non-ASCII characters (e.g., U+421).

**Comment by d1vanov** [comment]
Ok, I'll try to reproduce myself when I have some time.

**Comment by hosiet** [comment]
[document_build_error_snippet.txt](https://github.com/d1vanov/QEverCloud/files/516137/document_build_error_snippet.txt)

**Comment by d1vanov** [comment]
Any hint on how to make Doxygen use xelatex instead of pdflatex?
Currently Doxyfile generated by Doxygen seems to only offer pdflatex as an option:
```
# If the USE_PDFLATEX tag is set to YES, doxygen will use pdflatex to generate
# the PDF file directly from the LaTeX files. Set this option to YES, to get a
# higher quality PDF documentation.
# The default value is: YES.
# This tag requires that the tag GENERATE_LATEX is set to YES.

USE_PDFLATEX           = YES
```

**Comment by d1vanov** [comment]
Funny, it seems this particular error showed up because of the cyrillic `С` letter (Unicode symbol U+421) in the documentation for template `Optional` class, in `С++` word at line 20 of `Optional.h`. I can't verify it right now but most probably the replacement of that character to the one from the English alphabet should fix it.

**Comment by d1vanov** [comment]
As far as I can see, it's the only location of this particular Unicode symbol within the project.

**Comment by d1vanov** [comment]
Just to clarify, on a typical keyboard with Russian letters that cyrillic symbol and the English letter `C` are inputted via the same button, depending on your currently selected keyboard layout. So it's actually a frequent issue for people inputting something in both Russian and English 😃

**Comment by hosiet** [comment]
Well you can't always make sure there won't be any non-ASCII characters in the future :p

Although I can switch to xelatex,

```diff
diff --git a/Doxyfile.in b/Doxyfile.in
index 51eab28..b51ed1d 100644
--- a/Doxyfile.in
+++ b/Doxyfile.in
@@ -1604,7 +1604,7 @@ LATEX_OUTPUT           = latex
 # The default file is: latex.
 # This tag requires that the tag GENERATE_LATEX is set to YES.
 
-LATEX_CMD_NAME         = latex
+LATEX_CMD_NAME         = xelatex
 
 # The MAKEINDEX_CMD_NAME tag can be used to specify the command name to generate
 # index for LaTeX.
@@ -1703,7 +1703,7 @@ PDF_HYPERLINKS         = YES
 # The default value is: YES.
 # This tag requires that the tag GENERATE_LATEX is set to YES.
 
-USE_PDFLATEX           = YES
+USE_PDFLATEX           = NO
 
 # If the LATEX_BATCHMODE tag is set to YES, doxygen will add the \batchmode
 # command to the generated LaTeX files. This will instruct LaTeX to keep running
```
(from http://www.stack.nl/~dimitri/doxygen/manual/config.html#config_latex)

...`ps2pdf` would refuse to work without pdflatex. What a pity.

I am not that familiar with Doxygen, so maybe the best solution is to ensure full
ASCII characters within the source documentation.

**Comment by d1vanov** [comment]
Fixed in commit [bd527d1](https://github.com/d1vanov/QEverCloud/commit/bd527d1bff8fff7f04ba7203d7a51fbba66a62f7). 

Please note that I've raised the library version to 3.0.3 to properly account for the fixes for issue #7 and this one.
